// @sonar/scan/src/platform.js
"use strict";
/*
 * sonar-scanner-npm
 * Copyright (C) 2022-2025 SonarSource Sàrl
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getArch = getArch;
exports.isLinux = isLinux;
exports.isWindows = isWindows;
exports.isMac = isMac;
exports.getSupportedOS = getSupportedOS;
const deps_1 = require("./deps");
const logging_1 = require("./logging");
function getArch() {
    const { process } = (0, deps_1.getDeps)();
    return process.arch;
}
function isLinux() {
    const { process } = (0, deps_1.getDeps)();
    return process.platform === 'linux';
}
function isWindows() {
    const { process } = (0, deps_1.getDeps)();
    return process.platform === 'win32';
}
function isMac() {
    const { process } = (0, deps_1.getDeps)();
    return process.platform === 'darwin';
}
/**
 * @see https://github.com/microsoft/vscode/blob/64874113ad3c59e8d045f75dc2ef9d33d13f3a03/src/vs/platform/extensionManagement/common/extensionManagementUtil.ts#L171C1-L190C1
 */
function isAlpineLinux() {
    const { fs } = (0, deps_1.getDeps)();
    if (!isLinux()) {
        return false;
    }
    let content;
    try {
        const fileContent = fs.readFileSync('/etc/os-release');
        content = fileContent.toString();
    }
    catch (error) {
        try {
            const fileContent = fs.readFileSync('/usr/lib/os-release');
            content = fileContent.toString();
        }
        catch (error) {
            (0, logging_1.log)(logging_1.LogLevel.WARN, 'Failed to read /etc/os-release or /usr/lib/os-release');
        }
    }
    const match = /^ID=([^\r\n]*)/m.exec(content ?? '');
    return !!content && (match ? match[1] === 'alpine' : false);
}
function getSupportedOS() {
    const { process } = (0, deps_1.getDeps)();
    return isAlpineLinux() ? 'alpine' : process.platform;
}
